utils.ts

```diff
@@ -49,7 +49,7 @@ export function serialize(val: string): u32 {
   for (let i = 0; i < dataLength; ++i) {
     store<u16>(ptr + i, val.charCodeAt(i));
```

Comment:
Are you sure this should `u16` storage instead `u8` byte array?

Reply:
The purpose of serialize is to convert an assemblyscript encoded string (utf-18LE) to a utf-8 encoded string and returning the memory offset and size of the new string encoded in a single u32. 

As you suggested perhaps we should be using string.toUTF8 instead? It seems there are some special cases we are not handling

Reply:
AS used UTF-16LE encoding, yeah. And you need `u8` if avoid using UTF-8 and use ASCII. So you should use `store<u8>` instead `store<u16>`.

Reply:
So this should looks like:
 ```ts
for (.....) {
  store<u8>(ptr + i, <u8>(val.charCodeAt(i) & 0x00FF));
}
```

Reply:
Of use `String#toUTF8` which more right but may be you don't need uft-8 and only ASCII?

Reply:
I propose we trial using the string.toUTF8 approach. We aim to support the full JSON standard which specifies encoding MUST be in UTF-8, UTF-16, or UTF-32. There is no guarantee that only ASCII characters will be present.